OpenWeather/Networking/Implementation/LocationProvider.swift

```diff
@@ -55,10 +54,76 @@ struct LocationProvider: LocationProviding {
             }
         }
     }
+
+    func locations(for address: String) async throws -> [DeviceLocation] {
+        let response: LocationSearchResponse = try await locations(for: address)
+
+        return response.result
+            .map {
+                DeviceLocation(
+                    name: $0.name,
+                    location: $0.location
+                )
+            }
```

Comment:
Ideally we would have an adapter for this and we could make tests on the adapter. This is fine, but we should prefer adapters.. If either device or network have 3 or more variables I would say we require an adapter.. but this only has 2... if you plan on adding more data to these models, I'd say let's make an adapter and tests.

Reply:
I think that it would be smart to incorperate an adapter at this point in an effort to futureproof the code, and as you noted - will make it easier to test.